packages/react-app/components/cards/MediaCard.jsx

```diff
@@ -70,27 +72,55 @@ function MediaCard({
   const toast = useToast();
 
   const handleRequestPrivateProfileUnlock = async () => {
+    const stakeAmount = prompt("Enter stake amount in MATIC");
+    if (!stakeAmount) {
+      return alert("Please enter a valid stake amount");
+    }
+    const weiStakeAmount = ethers.utils.parseEther(stakeAmount);
     try {
-      const tx = await dRecruitContract.request(privateProfile.tokenId, {
-        value: ethers.utils.parseEther("0.01"),
+      // Request allowance
+      const dERC20Address = await dRecruitContract.token();
+      const signer = await context.injectedProvider.getSigner();
+      const dERC20Contract = new ethers.Contract(dERC20Address, abis.ERC20, signer);
```

Comment:
This process should not happen on the fly but stored in state on first load, similar to other contracts. You can put it as the last step of `loadWeb3Modal()`